src/lando/ui/legacy/forms.py

```diff
@@ -12,8 +15,32 @@ class TransplantRequestForm(forms.Form):
 class UpliftRequestForm(forms.Form):
     """Form used to request uplift of a stack."""
 
-    revision_id = forms.RegexField(regex="D[0-9]+$")
-    repository = forms.CharField()
+    revision_id = forms.RegexField(
+        regex="D[0-9]+$",
+        widget=forms.widgets.HiddenInput,
+        required=False,
+    )
+    repository = forms.ChoiceField(
+        widget=forms.Select(),
+        choices=((repo, repo) for repo in get_uplift_repositories()),
+    )
+
+    def clean_repository(self) -> str:
+        repo_name = self.cleaned_data["repository"]
+        all_repos = Repo.get_mapping()
+        repository = all_repos.get(repo_name)
```

Comment:
Does this only select upliftable repos?

Reply:
No, it will just verify that the repo is in fact a repo. I will clean up this section a little, but I think in the long run we should probably sync an "uplift" flag that is on the Repo object itself, and that way we can query these repos directly without contacting phabricator.

Reply:
Yep. It might be worth having a deliberate action or command to sync from Phabricator, but otherwise not do it, or do it periodically rather than on demand. Looking in suite, I saw that lando fires a large number of requests at Phabricator, and reducing them could potentially make the whole experience faster.